akc/provider.go

```diff
@@ -1,15 +1,41 @@
 package akc
 
 import (
-	"log"
-
+	"github.com/arkiaconsulting/terraform-provider-akc/client"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
 // Provider akc
 func Provider() *schema.Provider {
 	return &schema.Provider{
-		Schema: map[string]*schema.Schema{},
+		Schema: map[string]*schema.Schema{
+			"endpoint": {
+				Type:        schema.TypeString,
+				Required:    true,
+				DefaultFunc: schema.EnvDefaultFunc("AZAPPCONF_ENDPOINT", nil),
+			},
+			"client_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("ARM_CLIENT_ID", nil),
+			},
+			"client_secret": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Sensitive:   true,
+				DefaultFunc: schema.EnvDefaultFunc("ARM_CLIENT_SECRET", nil),
+			},
+			"tenant_id": {

```

Comment:
I think that the guidelines is now to have the tenant Id sensitive ( at least it is in the AzureCLI pipeline task in Azure DevOps)